### PR TITLE
Specify allowed contexts in jobs.<job_id>.services.<service_id>.image

### PR DESCRIPTION
**Pull request by piotrekkr**
### Why:

Documentation does not mention if and which context variable can be used in `jobs.<job_id>.services.<service_id>.image`. From my tests a subset of contexts can be used and those are: `github`, `matrix`, `strategy`, `needs`, `vars` and `inputs`. 

Unfortunately, using `env` context is not possible. Not sure why that is. Anyhow, specifying allowed contexts in docs would be helpful when writing workflows. 

I've also changed conditional service creation example because it contained invalid example that was using `options` context which does not exist.

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Added a note about allowed contexts and fixed example conditional service creation.

### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline (this link will be available after opening the PR).

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://docs.github.com/en/contributing/collaborating-on-github-docs/self-review-checklist).

### CODE REVIEW

**Comment by nguyenalex836** [comment]
@piotrekkr Thanks so much for opening a PR! I'll get this triaged for review ✨

File: content/actions/using-workflows/workflow-syntax-for-github-actions.md [hunk 1]
```diff
@@ -979,12 +979,16 @@ services:
 
 The Docker image to use as the service container to run the action. The value can be the Docker Hub image name or a registry name.
 
+> [!NOTE]
+> Usage of GitHub [AUTOTITLE](/actions/learn-github-actions/contexts) is allowed when specyfing service image.
+> Allowed contexts are: `github`, `strategy`, `matrix`, `needs`, `vars` and `inputs`.
```
Comment (hubwriter) [inline content/actions/using-workflows/workflow-syntax-for-github-actions.md:984]: I'd suggest rewording this slightly to use the active voice.

```suggestion
> You can use the following [AUTOTITLE](/actions/learn-github-actions/contexts) when specifying a service image: `github`, `strategy`, `matrix`, `needs`, `vars`, and `inputs`.
```

**Comment by hubwriter** [comment]
@piotrekkr - Thanks for raising this PR. I've asked the development team to confirm your changes before we go ahead and merge this.

File: content/actions/using-workflows/workflow-syntax-for-github-actions.md [hunk 2]
```diff
@@ -979,12 +979,15 @@ services:
 
 The Docker image to use as the service container to run the action. The value can be the Docker Hub image name or a registry name.
 
+> [!NOTE]
```
Comment (ericsciple) [inline content/actions/using-workflows/workflow-syntax-for-github-actions.md:982]: Is this a new pattern within this document? Wondering whether we need to duplicate this information here, since it's already [listed here](https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability)

File: content/actions/using-workflows/workflow-syntax-for-github-actions.md [hunk 3]
```diff
@@ -979,12 +979,15 @@ services:
...
+> You can use the following [AUTOTITLE](/actions/learn-github-actions/contexts) when specifying a service image: `github`, `inputs`, `matrix`, `needs`, `strategy`, and `vars`.
+
 If `jobs.<job_id>.services.<service_id>.image` is assigned an empty string, the service will not start. You can use this to set up conditional services, similar to the following example.
 
 ```yaml
 services:
   nginx:
-    image: {% raw %}${{ options.nginx == true && 'nginx' || '' }}{% endraw %}
+    image: {% raw %}${{ matrix.nginx_service == 'true' && 'nginx' || '' }}{% endraw %}
```
Comment (ericsciple) [inline content/actions/using-workflows/workflow-syntax-for-github-actions.md:990]: Wondering whether `matrix.nginx_service == true` would be better. Booleans are supported matrix values.

File: content/actions/using-workflows/workflow-syntax-for-github-actions.md (hunk 2 above)
Comment (piotrekkr) [reply content/actions/using-workflows/workflow-syntax-for-github-actions.md:982]: Wow, did not see this section before. I can see it was committed around a month ago so this is probably why. You are right. I'll drop this.

Comment (piotrekkr) [reply content/actions/using-workflows/workflow-syntax-for-github-actions.md:982]: Actually if allowed context are specified clearly already. I will close this PR.

**Comment by piotrekkr** [comment]
Closing this pull request because there is a list of available contexts in each configuration sections [here](https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability).
